### ISSUE TITLE
PermissionError on experiments checkpoint outputs 

### ISSUE THREAD

**Issue by geblanco** [iterative/dvc#5432]
# Bug Report

experiments run: checkpoint output is not writable, wrong tmp dir.

## Description

When running a stage containing an output as checkpoint, the script or command responsible for that stage writes to that file, because it is an output, but, with the new experiments functionality, as checkpoint outputs are stored in a temporary directory, that directory should be writeable. In the experiments wiki it is stated that tmp dir will be in `.dvc/tmp`, but it is not, it defaults to the system's directory, which is not writeable by default.

### Reproduce

```bash


set -e
# setup environment
mkdir ./virtualenvs
virtualenv ./virtualenvs/dvc_intro
source ./virtualenvs/dvc_intro/bin/activate
mkdir -p dvc_intro/data
wget https://gist.githubusercontent.com/geblanco/a906dd1f3ccaaa123f9ca28793a30d64/raw/2c27001194ed19841d3da1b1c34706bca97fbf20/dvc-getting-started-train-checkpoint.patch -O train.patch
# configure repo
cd dvc_intro
git init
dvc init
dvc config core.experiments true
dvc get https://github.com/iterative/dataset-registry get-started/data.xml -o data/data.xml
dvc add data/data.xml
git add data/data.xml.dvc data/.gitignore
wget https://code.dvc.org/get-started/code.zip
unzip code.zip
rm -f code.zip
pip install -r src/requirements.txt
pip install dvc
patch -p1 src/train.py ../train.patch
git add --all
git commit -am 'Add code + params'
# create stages
dvc push
dvc run -n prepare -p prepare.seed,prepare.split -d src/prepare.py -d data/data.xml -o data/prepared python src/prepare.py data/data.xml
dvc run -n featurize -p featurize.max_features,featurize.ngrams -d src/featurization.py -d data/prepared -o data/features python src/featurization.py data/prepared data/features
git add --all
git commit -am 'Add prepare + featurize stages'
dvc run -n train -p train.seed,train.n_estimators -d src/train.py -d data/features -c model.pkl --no-exec python src/train.py data/features model.pkl
dvc run -n evaluate -d src/evaluate.py -d model.pkl -d data/features -M scores.json --plots-no-cache prc.json --no-exec python src/evaluate.py model.pkl data/features scores.json prc.json
git add --all
git commit -m 'Add train/eval queued stages'
dvc push
dvc experiments run
```

### Expected

The script should be able to write the output file and continue writing it until it runs all experiment iterations. To do, dvc should employ a writeable tmp dir, for experiment commands (as stated in the wiki), or maybe let it be configured.

### Environment information

**Output of `dvc version`:**

```console
$ dvc version
1.11.15
```

**Additional Information (if any):**

**Comment by pmrowla** [comment]
Can you test using the current master branch from git? The wiki page for experiments reflects what's currently available in master, and the `1.11.x` releases are missing a couple of months worth of experiments changes which will not be backported (like the temp directory location change).

**Comment by geblanco** [comment]
Hello @pmrowla!

I can confirm that using the master branch, the error is avoided.

Thanks!
